### ISSUE TITLE
Android - Background State - Mic is not accessible

### ISSUE THREAD

**Issue by codoffer** [ZEGOCLOUD/zego_uikit_prebuilt_call_flutter#48]
- [x] I have read the [Getting Started](https://www.zegocloud.com/doc/callkit/Call%20Kit/main/Quick%20start/b26d509e) and [Getting Started(with call invitation)](https://www.zegocloud.com/doc/callkit/Call%20Kit/main/Quick%20start%20(with%20call%20invitation)/eb1d3c42)
- [x] I have already searched for the same problem

## Environment

| Technology            | Version       |
|-----------------------| ------------- |
| Flutter version      |       3.22.2      |
| Plugin version       |       ^4.12.1       |
| Android version      |    14          |
| iOS version          |        17.4      |
| macOS version        |              |
| Xcode version        |     15.4          |
| Google Chrome version |    125.0.6422.142          |

Device information: Realme 11 Pro Plus - Android 14 (and other Android devices also)

## Description

**Expected behavior:** 
A normal voice calls with all modes - foreground, background, and kill mode.

**Current behavior:** 

During voice calls, if the user (callee or caller) closes the app or locks the phone. The app is not able to access the microphone and the audio gets muted. As a result, there is no audio from one side.

## Steps to reproduce
Install with UI kit with basic flutter and check with the Android device (app in the background or locked status).

## Additional efforts
After reading multiple articles, we tried to use foreground services for Android. We used one package (https://pub.dev/packages/flutter_foreground_task) to keep the call mic status active. But all was in vain.

We are looking for help with the above package or any alternative way to enable the mic during the background/kill state.  What will be the service/function / method to be used with the above package to keep the mic accessible?

**Comment by yoer** [comment]
Please use foreground keep-alive. 
it cannot be embedded in UIKit, so you need to embed it yourself in your app.

**Comment by codoffer** [comment]
@yoer Thanks. Which service or method of zegocloud?

**Comment by yoer** [comment]
i mean, you can refer to plugins like [flutter_background](https://pub.dev/packages/flutter_background) or [flutter_foreground_task](https://pub.dev/packages/flutter_foreground_task).

**Comment by codoffer** [comment]
I have tried these plugins. These plugins require a specific task to run in the foreground. I am not sure, what will be the task or task to keep alive during background or kill state. Will we have to keep alive any specific zegocloud function?

**Comment by codoffer** [comment]
@yoer 
We are using only API for calling (https://www.zegocloud.com/docs/uikit/callkit-flutter/invitation-config/api-for-call-invitation). We have limited access to the calling screen with UI Kit. 

Using the above plugins - We can put a specific code/action in the foreground mode. Here, I do not have any action or widget to put in the foreground.  Please suggest more about this.

It's a major bug for the voice call feature.

**Comment by atulproject99** [comment]
@yoer please give your suggestion on this in detail so we can implemented this

**Comment by yoer** [comment]
You need to implement **app keeping alive** by yourself, we do not provide a keeping alive solution

**Comment by yoer** [comment]
this [example](https://github.com/ZEGOCLOUD/zego_uikit_prebuilt_live_audio_room_example_flutter/tree/master/live_audio_room) should be keep alive during background, You can refer to it

**Comment by codoffer** [comment]
@yoer Using this, we are getting another issue with the Android devices. The app keeps active with the locked screen and not getting any calls.

**Comment by yoer** [comment]
do you mean, the app is now kept alive in the background.
but when the screen is locked, the call cannot be received?

**Comment by codoffer** [comment]
Yes, I am aware of offline calls with background and killed state.

**Issue**
The user locks the phone during the app in the foreground stage. The app keeps opening and not able to receive notifications or calls. It may be due to the wake lock, etc.

**Comment by yoer** [comment]
Can you give me a log and the specific time of occurrence (call time)?

Regarding how to get the logs, you can refer to the **How ​​to get android device logs** chapter of [this document](https://zegocloud.feishu.cn/wiki/GOEfwJX11ifNF3kTFH8c57ownCf)

**Comment by codoffer** [comment]
@yoer Please share your email address. I will share zip files.

**Comment by yoer** [comment]
iamyoer@gmail.com

**Comment by codoffer** [comment]
@yoer We have tried with the updated document. It's crashing with Android 14.
